Let's start by importing the required module.

In [2]:
import numpy

Now, store the independent variables in X and the dependent variable in y.

In [3]:
#X represents the size of a tumor in centimeters.
X = numpy.array([3.78, 2.44, 2.09, 0.14, 1.72, 1.65, 4.92, 4.37, 4.96, 4.52, 3.69, 5.88]).reshape(-1,1)

#Note: X has to be reshaped into a column from a row for the LogisticRegression() function to work.
#y represents whether or not the tumor is cancerous (0 for "No", 1 for "Yes").
y = numpy.array([0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1])

As we shall use a method from the sklearn module,  we import that module as well.

In [4]:
from sklearn import linear_model

 We have used the LogisticRegression() method for creating the logistic regression object from the sklearn module.

This object comprises a method called fit() that takes the independent and dependent values as parameters by filling the regression object with data that describes the relationship.

In [5]:
logr = linear_model.LogisticRegression()
logr.fit(X,y)

LogisticRegression()

Here we have a logistic regression object that is ready for checking whether a tumor is cancerous based on the tumor size.

In [6]:
#predict if tumor is cancerous where the size is 3.46cm:
predicted = logr.predict(numpy.array([3.46]).reshape(-1,1))
print(predicted)

[0]


We predicted that a tumor with a size of 3.46cm is not cancerous.

In logistic regression, the coefficient is the expected change in log odds of having the outcome per unit change in X.

In [7]:
log_odds = logr.coef_
odds = numpy.exp(log_odds)

print(odds)

[[4.03541657]]


It implies that the size of a tumor increases by 1cm the odds of it being a tumor increases by 4x.

The coefficient and intercept values are used for finding the probability that each tumor is cancerous.

Let's create a function that uses the model's coefficient and intercept values for returning a new value. This new value represents probability that the given observation is a tumor:

In [8]:
def logit2prob(logr,x):
  log_odds = logr.coef_ * x + logr.intercept_    #formula analogous to linear regression for extracting the coefficient and the intercept.
  odds = numpy.exp(log_odds)                     #To convert the log-odds to odds we must exponentiate the log-odds.
  probability = odds / (1 + odds)                #Converting the odds to probability by dividing it by 1 plus the odds.
  return(probability)

print(logit2prob(logr, X))

[[0.60749955]
 [0.19268876]
 [0.12775886]
 [0.00955221]
 [0.08038616]
 [0.07345637]
 [0.88362743]
 [0.77901378]
 [0.88924409]
 [0.81293497]
 [0.57719129]
 [0.96664243]]


The output can be interpreted in the following way:

1. The probability that a tumor with the size 3.78cm is cancerous is 60.75% (approx).
2. The probability that a tumor with the size 2.44cm is cancerous is 19.269% (approx).
3. The probability that a tumor with the size 2.09cm is cancerous is 12.775% (approx).

and so on.